### PR TITLE
ls: use the spf13 flag package

### PR DESCRIPTION
**Pull request by rminnich**
Now we can do ls -lR, finally!

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/ls/ls.go [hunk 1]
```diff
@@ -17,8 +17,8 @@
 package main
 
 import (
-	"flag"
 	"fmt"
+	flag "github.com/spf13/pflag"
```
Comment (hugelgupf) [inline cmds/ls/ls.go]: super nit! please separate this from std lib imports

```go
import (
  "fmt"

   flag "github.com..."
)
```

Comment (rminnich) [reply cmds/ls/ls.go]: done. Now you need to do a fast review :-)

File: cmds/ls/ls.go [hunk 2]
```diff
@@ -17,21 +17,22 @@
...
 )
 
 var (
-	all     = flag.Bool("a", false, "show hidden files")
-	human   = flag.Bool("h", false, "human readable sizes")
-	long    = flag.Bool("l", false, "long form")
-	quoted  = flag.Bool("Q", false, "quoted")
-	recurse = flag.Bool("R", false, "equivalent to findutil's find")
+	all     = flag.BoolP("all", "a", false, "show hidden files")
+	human   = flag.BoolP("human", "h", false, "human readable sizes")
```
Comment (rjoleary) [inline cmds/ls/ls.go]: Can you use the GNU long names?

```
--human-readable
--quote-name
--recursive
```

Comment (rminnich) [reply cmds/ls/ls.go]: I hate the gnu names but you are right, I have to do it.

Comment (rminnich) [reply cmds/ls/ls.go]: done

**Pull request merged**